akd/tools/data_search/_base.py

```diff
@@ -0,0 +1,315 @@
+"""
+Base classes for data search tools.
+"""
+
+import json
+from typing import Any, Dict, Optional
+
+from pydantic import Field
+from pydantic.networks import HttpUrl
+
+from akd._base import InputSchema, OutputSchema
+from akd.tools._base import BaseTool, BaseToolConfig
+from akd.utils.logging import ContextualLogger
+
+
+class DataSearchToolConfig(BaseToolConfig):
```

Comment:
Let's base this off from `akd.tools.search` search config to maintain the semantics of "search"